ArticleMAJ.java

```diff
@@ -0,0 +1,158 @@
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.BorderFactory;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.JSpinner;
+
+public class ArticleMAJ extends JDialog {
+    private ToStringArticleMAJ infoToString = new ToStringArticleMAJ();
+    private boolean sendData;
+    private JLabel designationLabel, prixLabel, quantiteLabel, articleLabel;
+    private JComboBox article;
+    private JTextField designation;
+    private JSpinner prix, quantite;
+
+    BaseDeDonnees b;
+    String modification;
+
+    public ArticleMAJ(JFrame parent, String title, boolean modal, String modif, BaseDeDonnees bd) {
+        super(parent, title, modal);
+        this.setSize(530, 215);
+        this.setLocationRelativeTo(null);
+        this.setResizable(false);
+        this.setDefaultCloseOperation(JDialog.DO_NOTHING_ON_CLOSE);
+        b = bd;
+        modification = modif;
+        this.initComponent();
+    }
+
+    public ToStringArticleMAJ showArticleMAJ() {
+        this.sendData = false;
+        this.setVisible(true);
+        return this.infoToString;
+    }
+
+    private void initComponent() {
+
+        JPanel content = new JPanel();
+
+        // Article
+        JPanel panArticle = new JPanel();
+        panArticle.setBackground(Color.white);
+        panArticle.setPreferredSize(new Dimension(480, 60));
+        panArticle.setBorder(BorderFactory.createTitledBorder("Selection de l'article a modifier"));
+        article = new JComboBox();
+        if (b.getListeArticles().length == 0 || b.getListeArticles() == null) {
+            article.addItem("Pas d'article a modifier");
+        } else {
+            article.addItem("");
+            for (int i = 0; i < b.getListeArticles().length; i++) {
+                article.addItem("reference : " + b.getListeArticles()[i].getNumero() + " - "
+                        + b.getListeArticles()[i].getDesignation() + " - " + b.getListeArticles()[i].getPrix()
+                        + " euro(s) " + b.getListeArticles()[i].getQuantite() + " en stock");
+            }
+        }
+
+        articleLabel = new JLabel("");
+        panArticle.add(articleLabel);
+        panArticle.add(article);
+        content.setBackground(Color.white);
+        content.add(panArticle);
+
+        if (modification == "designation") {

```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.